### DISCUSSION TITLE
uosc's compatibility with other scripts.

### DISCUSSION THREAD

**Discussion by WhoMI7** [tomasklaen/uosc#496] [General]
Some scripts have the tendency to break uosc's functionality, example: can't click on buttons, volume slider or uosc menu.
Is there any particular reason why? 

Here is an example: [MPV-Watch-Later-OSD-Menu](https://github.com/JKubovy/MPV-Watch-Later-OSD-Menu), if I add this to my scripts, and scroll on timeline _the volume changes_ with volume bar hidden.
And if I delete the script, it works just fine.

**Comment by OmiChrome** [comment]
i have experienced that too 
uosc getting broken

**Comment by christoph-heinrich** [comment]
Are you on the latest uosc commit? There has been a similar issue recently https://github.com/tomasklaen/uosc/issues/495

**Reply by WhoMI7** [reply]
Yeah. But maybe being on entirely different osc might be the problem. Like [JKubovy](https://github.com/JKubovy/MPV-Watch-Later-OSD-Menu/issues/1#issuecomment-1496980286) here said.
And if not using default osc is the problem then using any script which was written without having uosc as reference will not be compatible. Ultimately making uosc incompatible with every other script out there.

**Reply by christoph-heinrich** [reply]
I've tried installing that script and uosc still works fine for me. I've also had a glance at it's source code and didn't see anything that could possibly cause that.
FWIW I've never had any compatiblity problems between uosc and other scripts before.

**Comment by tomasklaen** [comment]
I've skimmed through the source code of that script and haven't noticed anything obvious.

Are you sure you are on the latest uosc? There was an issue where using `autohide` would un-register cursor events listeners after a while.

**Reply by alexpandain** [reply]
github release has old script. I git clone to get latest script

**Reply by WhoMI7** [reply]
> Are you sure you are on the latest uosc? There was an issue where using `autohide` would un-register cursor events listeners after a while.

Updated again just to confirm and issue persists.

https://user-images.githubusercontent.com/93109835/230104068-00e8fb38-5000-4c9a-b1aa-e1f2cbeb8739.mp4


> I've skimmed through the source code of that script and haven't noticed anything obvious.

If this script isn't breaking uosc _directly_ then it is acting as a catalyst.
here's my scripts folder 
[scripts.zip](https://github.com/tomasklaen/uosc/files/11159740/scripts.zip)

If I delete `webm.lua` this issue ceases to exist. But that doesn't really mean this alone was the reason?
So onto my original question, do scripts have to be written with uosc kept as reference to be compatible?

**Reply by christoph-heinrich** [reply]
> do scripts have to be written with uosc kept as reference to be compatible?

No

**Reply by tomasklaen** [reply]
It was another `autohide` bug 😐. Cursor events were not refreshing already started autohide timer. Update to main branch, should be fixed.

**Reply by tomasklaen** [reply]
Damn, forgot to switch back to my scripts folder before testing and pushed buggy commit. **Now** it's fixed.

**Reply by WhoMI7** [reply]
> Damn, forgot to switch back to my scripts folder before testing and pushed buggy commit. **Now** it's fixed.

Still the same.

**Reply by hooke007** [reply]
> Damn, forgot to switch back to my scripts folder before testing and pushed buggy commit. Now it's fixed.

https://github.com/tomasklaen/uosc/commit/8d85068b7c457c8e17599a350ecbb530988189f1 It introduced the new bug here when clicking on the timeline.
https://user-images.githubusercontent.com/41094733/230272026-ddfad3fb-76b0-40f7-812d-50d44405dee7.mp4

**Reply by tomasklaen** [reply]
@BhaturaGuy I was referring to autohide triggering when scrolling speed, volume, etc, but I see that dragging timeline still had some leftover autohides. I've pushed another commit that should fix it all hopefully.

@hooke007 that was timeline speed seeking related optimizations that were there before, but I've tweaked those too.

**Reply by WhoMI7** [reply]
> I've pushed another commit that should fix it all hopefully

Updated just now, issue remains the same.

**Reply by tomasklaen** [reply]
Than you'll have to provide steps to reproduce, potentially mpv config directory, because I can't replicate any of the issues on your video.

Scrolling all speed, timeline, and menu works and doesn't autohide or stop working suddenly anymore if you keep scrolling without moving the cursor.

**Reply by WhoMI7** [reply]
Sure here's my config directory 
[mpv.zip](https://github.com/tomasklaen/uosc/files/11182732/mpv.zip)
 Steps to reproduce.

1. Backup your config then replace with mine.
2. Play any file.
3. Try to scroll up or down on speed button.

**Reply by tomasklaen** [reply]
Ok, you'll need to isolate it as well. You have like all the scripts and shaders in existence there. Include only the bare minimum that reproduces the issue.

**Reply by WhoMI7** [reply]
> Ok, you'll need to isolate it as well. You have like all the scripts and shaders in existence there. Include only the bare minimum that reproduces the issue.

I tried isolating but well issue just disappears if I delete either of the following scripts. And also when all the other scripts except them are deleted (uosc is excluded of course)
```
autosubsync
subselect
change-refresh
crop
dynamic-crop
evafast
inputevent
lrc
metadata_osd
quality-menu
recentmenu
save-playlist
simplebookmark
simplehistory
slicing_copy
smartcopypaste
undoredo
watch_later_osd_menu
webm
youtube-download
youtube-upnext
```
I copied my config from my PC (Windows 11) to laptop (Windows 10) and the results were same "no scroll feature i.e. no scroll on speed button, timeline, uosc menu."
Did you try to copy paste my above config? Cause if you can't reproduce the issue then maybe the issue is on my end (PC and laptop both?).

**Reply by hooke007** [reply]
Apparently you manually create this bug.
```ini
# check your conf
key_moveup=UP Wheel_Up
key_movedown=DOWN Wheel_Down
key_movepageup=PGUP
key_movepagedown=PGDWN
key_movebegin=HOME
key_moveend=END
key_playfile=ENTER MBTN_LEFT
```
Different scripts use the same dynamic/force keybindings at the same time.
I have seen you were using a complex and hacky binding system while there are plenty of keybindings you had set (or created by those scripts). So it's easy to meet with keybinding-conflicts.

**Reply by WhoMI7** [reply]
> Different scripts use the same dynamic/force keybindings at the same time.

Aren't they activated only when their respective menu is active?
Can't see them replacing keybinds in `Active key bindings` until their menu is activated?
But if it is the only possible reason, I can change the keybinds but to what extent? Just from the scripts-opts or `input.conf` too?
And is using `Wheel_UP`, `Wheel_DOWN` for changing volume, scrolling menus (uosc or other) such a bad idea?
And let's say I use `Wheel_UP` and `Wheel_DOWN` for volume, `;` and `'` for menu scrolling, wouldn't this be actually complex?

**Reply by hooke007** [reply]
> I have seen you were using a complex and hacky binding system

Here is the keypoint. Only use these two scripts and then no issues. It must be something else make them work at the same time.

**Reply by OmiChrome** [reply]
> Only use these two scripts and then no issues

this litrallyl breakes the usage of mpv as we use it 
the hackable player

**Reply by hooke007** [reply]
Cannot get your point. If you only need the functions the two scripts provided, then just use it. Nothing unreasonable here.
